code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -31,40 +33,32 @@
 	adjust_stagger(max(0, ex_slowdown - 2)) //Stagger 2 less than slowdown
 
 	//Sunder
-	var/sunder_loss = 50*(3 - severity) * bomb_sunder_multiplier
-	adjust_sunder(max(0, sunder_loss))
+	adjust_sunder(max(0, 50 * (3 - severity) * bomb_sunder_multiplier))
 
 	//Damage
 	var/ex_damage = 40 + rand(0, 20) + 50*(4 - severity)  //changed so overall damage stays similar
-	ex_damage *= bomb_damage_multiplier
-
-	apply_damage(ex_damage/2, BRUTE, updating_health = TRUE)
-	apply_damage(ex_damage/2, BURN, updating_health = TRUE)
+	apply_damages(ex_damage * 0.5, ex_damage * 0.5, blocked = BOMB, updating_health = TRUE)
 
 
-/mob/living/carbon/xenomorph/apply_damage(damage = 0, damagetype = BRUTE, def_zone, blocked = 0, sharp = FALSE, edge = FALSE, updating_health = FALSE)
-	if(status_flags & (GODMODE))
+/mob/living/carbon/xenomorph/apply_damage(damage = 0, damagetype = BRUTE, def_zone, blocked = 0, sharp = FALSE, edge = FALSE, updating_health = FALSE, penetration)
+	if(status_flags & (GODMODE) ||damagetype != (BRUTE||BURN))
 		return
-	var/hit_percent = (100 - blocked) * 0.01
-
-	if(hit_percent <= 0) //total negation
-		return 0
-
-	damage *= CLAMP01(hit_percent) //Percentage reduction
+	if(isnum(blocked))
+		damage -= clamp(damage * (blocked - penetration) * 0.01, 0, damage)
+	else
+		damage = modify_by_armor(damage, blocked, penetration, def_zone)
 
 	if(!damage) //no damage
 		return 0
 
-	//We still want to check for blood splash before we get to the damage application.
-	var/chancemod = 0
-	if(sharp)
-		chancemod += 10
-	if(edge) //Pierce weapons give the most bonus
-		chancemod += 12
-	if(def_zone != "chest") //Which it generally will be, vs xenos
-		chancemod += 5
-
 	if(damage > 12) //Light damage won't splash.
+		var/chancemod = 0
```

Comment:
I think this whole logic should be in check blood splash directly

Reply:
I didn't actually check this proc. It already does chance calc there, fucking why is this not all in there, guh.